Frontend/default.nix

```diff
@@ -1,10 +1,25 @@
+{ inputs, ... }:
 {
-  perSystem = {
-    treefmt.config = {
-      # Suppress autoformatting of frontend dhall files.
-      settings.formatter.dhall.excludes = [
-        "Frontend/packages.dhall"
-      ];
+  imports = [
+  ];
+  perSystem = { config, self', pkgs, lib, ... }:
+    let easy-ps = import inputs.easy-purescript-nix { inherit pkgs; };
+    in {
+      treefmt.config = {
+        # Suppress autoformatting of frontend dhall files.
+        settings.formatter.dhall.excludes = [
+          "Frontend/packages.dhall"
+        ];
+      };
+      devShells.frontend = pkgs.mkShell {
+        name = "ps-dev-shell";
+        packages = [
+          easy-ps.purs-0_15_4
+          easy-ps.spago
+          easy-ps.psa
+          pkgs.dhall
+          pkgs.nodejs-14_x
+        ];
+      };
```

Comment:
@belevy You want to add the following to this devshell:

```nix
inputsFrom = [ config.pre-commit.devShell ];
```

This will enable pre-commit hooks, and there will be no need for the frontend people to run `treefmt` (from Haskell devshell).